build.gradle.kts

```diff
@@ -189,29 +189,53 @@ allprojects {
             showStackTraces = true
             exceptionFormat = org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
         }
+
+        // this is the kotlin equivalent of a Groovy's "afterSuite" Closure to
+        // print a quick test summary.
+        // inspirations taken from https://stackoverflow.com/a/36130467/7079724 and
+        // https://github.com/gradle/kotlin-dsl-samples/issues/836#issuecomment-384206237
+        addTestListener(object : TestListener {
```

Comment:
The main build file is getting a bit too big IMO. I suggest extracting some logic out of the build file into custom plugins/tasks for better maintainability. Similar to what we did with the [DependencyRulesPlugin](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/buildSrc/src/main/java/org/eclipse/dataspaceconnector/gradle/DependencyRulesPlugin.java). What about adding this summary as a plugin in a separate story?

Reply:
yup that makes sense. more stuff is incoming (#1147), so best to establish precedent soon :)